ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryBuilder.java

```diff
@@ -493,35 +492,23 @@ private SharedKafkaStreamsRuntime getKafkaStreamsInstance(
     }
     final SharedKafkaStreamsRuntime stream;
     final KsqlConfig ksqlConfig = config.getConfig(true);
-    final String queryPrefix = ksqlConfig
-        .getString(KsqlConfig.KSQL_PERSISTENT_QUERY_NAME_PREFIX_CONFIG);
-    final String serviceId = ksqlConfig.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG);
     if (real) {
       stream = new SharedKafkaStreamsRuntimeImpl(
           kafkaStreamsBuilder,
-          config.getConfig(true).getInt(KsqlConfig.KSQL_QUERY_ERROR_MAX_QUEUE_SIZE),
+          ksqlConfig.getInt(KsqlConfig.KSQL_QUERY_ERROR_MAX_QUEUE_SIZE),
```

Comment:
Just some minor cleanup to use the `ksqlConfig` variable everywhere and more importantly, factor the shared runtime id logic into a single builder method in `QueryApplicationId` (where we currently have the method for building regular application ids)